crates/hyperqueue/src/server/autoalloc/process.rs

```diff
@@ -372,12 +387,68 @@ async fn perform_submits(
         return Ok(());
     }
 
-    // Ask the scheduler how many workers it needs
+    // TODO: do not run the code below if all queues have full backlog anyway
     let queries: Vec<WorkerTypeQuery> = queues
         .iter()
         .map(|(_, queue)| create_queue_worker_query(queue))
         .collect();
-    let response = senders.server.new_worker_query(&queries)?;
+    let responses = compute_query_responses(senders, queries)?;
+
+    // Now schedule the workers into the individual queues
+    let queue_ids: Vec<QueueId> = queues.into_iter().map(|(id, _)| id).collect();
+    for (response, &queue_id) in responses.into_iter().zip(&queue_ids) {
+        queue_try_submit(autoalloc, queue_id, senders, response).await;
+    }
+
+    for (queue_id, queue) in autoalloc.queues_mut() {
+        try_pause_queue(queue, queue_id);
+    }
+
+    Ok(())
+}
+
+/// Create a worker query that corresponds to the provided resources of the given `queue`.
+fn create_queue_worker_query(queue: &AllocationQueue) -> WorkerTypeQuery {
+    let (descriptor, partial) = {
+        if let Some(worker_config) = queue.get_worker_config() {
+            // If we have a known worker config, we estimate exact resources based on it
+            (worker_config.resources.clone(), false)
+        } else if let Some(resources) = queue.info().cli_resource_descriptor() {
+            // If not, try to at least estimate partial resources based from the CLI arguments
+            (resources.clone(), true)
+        } else {
+            // Otherwise just assume that the worker will have at least a single CPU
+            (
+                ResourceDescriptor::new(vec![ResourceDescriptorItem {
+                    name: CPU_RESOURCE_NAME.to_string(),
+                    kind: ResourceDescriptorKind::regular_sockets(1, 1),
```

Comment:
This is wrong logic. It says, "we do not know all resources about worker, but we know that it has 1 cpu" so it will be ignored by tasks that needs more cpus. 
If we really do no know nothing, then it should be an empty vec.
I other words, the partial resources are interpreted as lowebound on what we know about the worker, not lowerbound on resources.

Reply:
Good catch! I implemented this while still thinking in the "old regime" where we didn't have the lower bound.

However, it looks like worker query computation doesn't handle an empty vec currently:
```
called `Result::unwrap()` on an `Err` value: cannot compute worker query

Caused by:
    Resource 'cpus' is missing
```
It validates all the resource descriptors at start, and validation doesn't allow missing CPUs.